src/main/java/org/openpnp/machine/reference/vision/ReferenceBottomVision.java

```diff
@@ -65,10 +68,77 @@ public PartAlignmentOffset findOffsets(Part part, BoardLocation boardLocation,
 
         // Pre-rotate to minimize runout
         double preRotateAngle = 0;
-        if (preRotate && boardLocation != null && placementLocation != null) {
-            preRotateAngle =
-                    Utils2D.calculateBoardPlacementLocation(boardLocation, placementLocation)
-                           .getRotation();
+        if (preRotate) {
+			if(part == null || nozzle.getPart() == null) 					   { throw new Exception("no part on nozzle"); }			
+			if((!part.toString().equals(nozzle.getPart().toString()))) { throw new Exception("Part mismatch with part on nozzle"); }
+			part=nozzle.getPart();
+			double angle = placementLocation.getRotation();
+			if(boardLocation!=null) {
+				angle = Utils2D.calculateBoardPlacementLocation(boardLocation, placementLocation).getRotation();              
+			}
+			while (Math.abs(angle) > 180.0) { angle += (angle < 0.)? 360. : -360.; } 
+			double nozzleAngle=angle;
+			double placementAngle=angle;
+            MovableUtils.moveToLocationAtSafeZ(nozzle, VisionUtils.getBottomVisionCamera().getLocation().add(new Location(part.getHeight().getUnits(), 0.0, 0.0,part.getHeight().getValue(), 0.0)).derive(null, null, null, angle));
+            CvPipeline pipeline = partSettings.getPipeline();
+            pipeline.setCamera(VisionUtils.getBottomVisionCamera());
+            pipeline.setNozzle(nozzle);
+			pipeline.setValue(angle);
+			if(Boolean.getBoolean("develop")) {
+            if ("Rotate".equals(pipeline.getStages().get(1).getClass().getSimpleName())) {
+                ((Rotate)pipeline.getStages().get(1)).setDegrees( angle );
+            }}
+			pipeline.process();
+			if (!((pipeline.getResult("result")).model instanceof RotatedRect)) {
+				throw new Exception("Bottom vision alignment failed for part " + part.getId()
+						+ " on nozzle " + nozzle.getName() + ". No result found.");
+			}
+
+
+			RotatedRect rect = ((RotatedRect)(pipeline.getResult("result")).model);
+			if (rect.size.width < rect.size.height) { rect.angle = 90 + rect.angle; }
+            Logger.debug("Result rect {}", rect);
+			while (Math.abs(angle) > 45.0) { nozzleAngle= (angle += (angle < 0.)? 90 : -90); } 
+			while (Math.abs(rect.angle) > 45.0) { rect.angle += (rect.angle < 0)? 90 : -90; } 
+			angle += rect.angle;
+			while (Math.abs(angle) > 45.0) { angle += (angle < 0.)? 90 : -90; } 
+
+			int cnt=1;
+			if (pipeline.getResult("contours")!=null&&pipeline.getResult("contours").model instanceof List<?>) {
+				cnt=((List<?>)(pipeline.getResult("contours")).model).size();
+			}
+
+			String s="";
+			Location offsets = null;
+			int old=999;
```

Comment:
it can be replaced with 99 or any other value bigger as that. Ideally it should be greater as 360 and
for security greater as 720, so 999 fit nicely. Max_Integer fit too. The other requirement is that variable rot need to be initialized with the same value. As maxInt when converted to double is Infinite, i prefer safer value 999.  It is used inside old=Math.min(old,rot); and no , it is not a special
case  code, 765 or 123 works equals.